subsys/bluetooth/host/gatt.c

```diff
@@ -2214,7 +2214,8 @@ static void gatt_subscription_remove(struct bt_conn *conn, sys_snode_t *prev,
 	/* Remove subscription from the list*/
 	sys_slist_remove(&subscriptions, prev, &params->node);
 
-	params->notify(conn, params, NULL, 0);
+	if (conn)
+		params->notify(conn, params, NULL, 0);
```

Comment:
This one Im not really following, don't you want to notify that a subscription has been removed even if it is a result of unpair?

Reply:
`bt_gatt_clear` is called after `bt_conn_disconnect` (see `bt_unpair` in `hci_core.c`) - I guess that needs to be changed in order to fix this.

Reply:
Id say we should indicate in the documentation that conn can be NULL since unpair don't require one so that we can remove the subscription even when disconnected.

Reply:
Fixed.